app/handlers/newflow/educator_signup/complete_profile.rb

```diff
@@ -98,45 +121,69 @@ def format_books_for_salesforce_string(books)
         books.reject(&:empty?)&.join(';')
       end
 
+      def books_used
+        signup_params.books_used.reject{ |b| b.blank? }
+      end
+
+      def books_of_interest
+        signup_params.books_of_interest.reject{ |b| b.blank? }
+      end
+
       def check_params
-        if (signup_params.is_school_not_supported_by_sheerid == 'true' ||
-          signup_params.is_country_not_supported_by_sheerid == 'true') &&
-           signup_params.school_name.blank?
+        role = signup_params.educator_specific_role.strip.downcase
+        @did_use_sheerid = !signup_params.is_school_not_supported_by_sheerid == 'true' || !signup_params.is_country_not_supported_by_sheerid == 'true' || !user.is_sheerid_unviable?
 
+
+        if (!@did_use_sheerid) && signup_params.school_name.nil?
           param_error(:school_name, :school_name_must_be_entered)
         end
 
-        if signup_params.educator_specific_role.strip.downcase == OTHER &&
-           signup_params.other_role_name.blank?
-
+        if role == OTHER && signup_params.other_role_name.nil?
           param_error(:other_role_name, :other_must_be_entered)
         end
 
-        if signup_params.educator_specific_role.strip.downcase  == INSTRUCTOR &&
-           signup_params.using_openstax_how == AS_PRIMARY && books_used.blank?
-
+        if role  == INSTRUCTOR && signup_params.using_openstax_how == AS_PRIMARY && books_used.blank?
           param_error(:books_used, :books_used_must_be_entered)
         end
 
-        if signup_params.educator_specific_role.strip.downcase  == INSTRUCTOR &&
-           signup_params.using_openstax_how != AS_PRIMARY && books_of_interest.blank?
-
+        if role  == INSTRUCTOR && signup_params.using_openstax_how != AS_PRIMARY && books_of_interest.blank?
           param_error(:books_of_interest, :books_of_interest_must_be_entered)
         end
 
-        if signup_params.educator_specific_role.strip.downcase  == INSTRUCTOR &&
-          signup_params.num_students_per_semester_taught.blank?
-
+        if role  == INSTRUCTOR && signup_params.num_students_per_semester_taught.blank?
           param_error(:num_students_per_semester_taught, :num_students_must_be_entered)
         end
+
+        # if (!@did_use_sheerid || role != OTHER) && !signup_params.school_issued_email.blank?
```

Comment:
I assume this commented out code is here for a reason.

Reply:
It is, the email field on the CS version of the form is not being validated and I need to revisit it.

Reply:
This has been fixed. Here is the commit: https://github.com/openstax/accounts/pull/1028/commits/88c987a7264458351ac7359d3cc995a95ca02b70